diff --git a/index.html b/index.html
index 50f2963..58d9353 100644
--- a/index.html
+++ b/index.html
@@ -245,6 +245,7 @@
 	let csskeys = {};
 
 	const fs = require("fs");
+	const path = require("path");
 
 	$`#injecttoggle`.checked = localStorage.getItem('injectcss') == 'true'
 	$`#locktabs`.checked = localStorage.getItem('tablock') == 'true'
@@ -326,14 +327,14 @@
 				console.log("DTabs did not find an update. Upstream Version:",updatecontents.match(verregex)[1])
 			}
 		})
-	}	
-	
+	}
+
 	function update(){
         $`.updatedialog`.classList.add("active")
         $`.updatedialogbody`.setAttribute("style","display:none;")
         updatecheck(true);
-		fs.renameSync("./resources/app/index.html","./resources/app/index.html.OLD-"+Date.now());
-		fs.writeFileSync("./resources/app/index.html",updatecontents,"utf8");
+		fs.renameSync(path.resolve(__dirname, "index.html"), path.resolve(__dirname, "index.html.OLD-"+Date.now()));
+		fs.writeFileSync(path.resolve(__dirname, "index.html"), updatecontents,"utf8");
 		alert("Update has been applied, DTabs will now exit.");
 		window.close();
 	}
diff --git a/main.js b/main.js
index 2548f55..1b79670 100644
--- a/main.js
+++ b/main.js
@@ -1,5 +1,6 @@
 const { app, BrowserWindow, session,shell,dialog} = require('electron')
 const fs = require("fs");
+const path = require("path");
 
 let mainWindow
 
@@ -16,9 +17,9 @@ function createWindow() {
   mainWindow.on('closed', function () {
     mainWindow = null
   })
-  
-  if(!fs.existsSync("./resources/app/index.html")){
-    var files = fs.readdirSync("./resources/app");
+
+  if(!fs.existsSync(path.resolve(__dirname, "index.html"))) {
+    var files = fs.readdirSync(__dirname);
     var times = [];
     var timeextractor = /index.html.OLD-(\d+)/;
     for(var name of files){
@@ -34,7 +35,7 @@ function createWindow() {
       return;
     }
     times.sort((a,b)=>{return b-a;});
-    fs.renameSync("./resources/app/index.html.OLD-"+times[0],"./resources/app/index.html")
+    fs.renameSync(path.resolve(__dirname, "index.html.OLD-"+times[0]), path.resolve(__dirname,"index.html"))
     dialog.showMessageBox({type:"warning",title:"DTabs Warning",
         message:"WARNING: index.html was missing but DTabs found and reverted an update!\nThis may indicate a failed update or install corruption.\nIf you experience any issues reinstall DTabs!"})
   }
